integrations/ultralytics/deepsparse/deepsparse_utils.py

```diff
@@ -70,28 +65,50 @@ def load_image(
     return img
 
 
-def pre_nms_postprocess(outputs: List[numpy.ndarray]) -> torch.Tensor:
+class YoloPostprocessor:
     """
-    :param outputs: raw outputs of a YOLOv3 model before anchor grid processing
-    :return: post-processed model outputs without NMS.
+    Class for performing postprocessing of YOLOv3 model predictions
     """
-    # postprocess and transform raw outputs into single torch tensor
-    processed_outputs = []
-    for idx, pred in enumerate(outputs):
-        pred = torch.from_numpy(pred)
-        pred = pred.sigmoid()
 
-        # get grid and stride
-        grid = _YOLO_V3_GRIDS[idx]
-        anchor_grid = _YOLO_V3_ANCHOR_GRIDS[idx]
-        stride = 640 / _YOLO_V3_OUTPUT_SHAPES[idx]
-
-        # decode xywh box values
-        pred[..., 0:2] = (pred[..., 0:2] * 2.0 - 0.5 + grid) * stride
-        pred[..., 2:4] = (pred[..., 2:4] * 2) ** 2 * anchor_grid
-        # flatten anchor and grid dimensions -> (bs, num_predictions, num_classes + 5)
-        processed_outputs.append(pred.view(pred.size(0), -1, pred.size(-1)))
-    return torch.cat(processed_outputs, 1)
+    def __init__(self, image_size: Tuple[int]):
+        self._image_size = image_size
+        self._grids = {}  # Dict[Tuple[int], torch.Tensor]
+
+    def pre_nms_postprocess(self, outputs: List[numpy.ndarray]) -> torch.Tensor:
+        """
+        :param outputs: raw outputs of a YOLOv3 model before anchor grid processing
+        :return: post-processed model outputs without NMS.
+        """
+        # postprocess and transform raw outputs into single torch tensor
+        processed_outputs = []
+        for idx, pred in enumerate(outputs):
+            pred = torch.from_numpy(pred)
+            pred = pred.sigmoid()
+
+            # get grid and stride
+            grid_shape = pred.shape[2:4]
+            grid = self._get_grid(grid_shape)
+            anchor_grid = _YOLO_V3_ANCHOR_GRIDS[idx]
+            stride = self._image_size[0] / grid_shape[0]
+
+            # decode xywh box values
+            pred[..., 0:2] = (pred[..., 0:2] * 2.0 - 0.5 + grid) * stride
+            pred[..., 2:4] = (pred[..., 2:4] * 2) ** 2 * anchor_grid
+            # flatten anchor and grid dimensions -> (bs, num_predictions, num_classes + 5)
+            processed_outputs.append(pred.view(pred.size(0), -1, pred.size(-1)))
+        return torch.cat(processed_outputs, 1)
+
+    def _get_grid(self, grid_shape: Tuple[int]) -> torch.Tensor:
```

Comment:
NIT: move private method to after public ones

Reply:
this is at the bottom of the functions for this class

Reply:
ah yep, looked at the file, was hard to see the indent level for the following function :)